### PR TITLE
provider: Support default tags (resources aws_e* 5)

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Reference: #7926

Output from acceptance testing: Will be handled via CI, review, and daily acceptance testing

### CODE REVIEW

File: aws/resource_aws_elasticache_cluster.go [hunk 1]
```diff
@@ -252,80 +252,86 @@ func resourceAwsElasticacheCluster() *schema.Resource {
...
+					if v, ok := diff.GetOk("engine"); !ok || v.(string) == "redis" {
+						return nil
+					}
+					if _, ok := diff.GetOk("final_snapshot_identifier"); !ok {
+						return nil
+					}
+					return errors.New(`engine "memcached" does not support final_snapshot_identifier`)
+				},
+			),
+			SetTagsDiff,
```
Comment (YakDriver) [inline aws/resource_aws_elasticache_cluster.go:326]: Is this the correct way to do this?

Comment (anGie44) [reply aws/resource_aws_elasticache_cluster.go:326]: i'm not sure either 🙃 i've seen just the customdiff.Sequence pattern, where inside you can add `SetTagsDiff`, but i'll double check

File: website/docs/r/elasticache_cluster.html.markdown [hunk 2]
```diff
@@ -134,13 +134,14 @@ SNS topic to send ElastiCache notifications to. Example:
...
 
-* `tags` - (Optional) A map of tags to assign to the resource
+* `tags` - (Optional) A map of tags to assign to the resource. If configured with a provider [`default_tags` configuration block](https://www.terraform.io/docs/providers/aws/index.html#default_tags-configuration-block) present, tags with matching keys will overwrite those defined at the provider-level.
 
 ## Attributes Reference
 
 In addition to all arguments above, the following attributes are exported:
 
 * `arn` - The ARN of the created ElastiCache Cluster.
+* `tags_all` - A map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://www.terraform.io/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/elasticache_cluster.html.markdown:144]: Nit: can be moved down a bit to maintain alphabetical order

File: website/docs/r/elasticache_replication_group.html.markdown [hunk 3]
```diff
@@ -196,6 +196,7 @@ In addition to all arguments above, the following attributes are exported:
 * `cluster_enabled` - Indicates if cluster mode is enabled.
 * `configuration_endpoint_address` - The address of the replication group configuration endpoint when cluster mode is enabled.
 * `primary_endpoint_address` - (Redis only) The address of the endpoint for the primary node in the replication group, if the cluster mode is disabled.
+* `tags_all` - A map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://www.terraform.io/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/elasticache_replication_group.html.markdown:199]: similar nit as above

File: website/docs/r/elasticsearch_domain.html.markdown [hunk 4]
```diff
@@ -321,6 +321,7 @@ In addition to all arguments above, the following attributes are exported:
 * `endpoint` - Domain-specific endpoint used to submit index, search, and data upload requests.
 * `kibana_endpoint` - Domain-specific endpoint for kibana without https scheme.
 * `vpc_options.0.availability_zones` - If the domain was created inside a VPC, the names of the availability zones the configured `subnet_ids` were created inside.
+* `tags_all` - A map of tags assigned to the resource, including those inherited from the provider [`default_tags` configuration block](https://www.terraform.io/docs/providers/aws/index.html#default_tags-configuration-block).
```
Comment (anGie44) [inline website/docs/r/elasticsearch_domain.html.markdown:324]: similar alphabetical order nit

**Review by anGie44 — APPROVED** [review]
Small nits, otherwise LGTM once we figure out the customdiff usage 😄

File: aws/resource_aws_elasticache_replication_group.go [hunk 5]
```diff
@@ -291,34 +292,38 @@ func resourceAwsElasticacheReplicationGroup() *schema.Resource {
 			Update: schema.DefaultTimeout(waiter.ReplicationGroupDefaultUpdatedTimeout),
 		},
 
-		CustomizeDiff: customdiff.Sequence(
-			func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
-				if v := diff.Get("multi_az_enabled").(bool); !v {
+		CustomizeDiff: customdiff.All(
+			customdiff.Sequence(
```
Comment (anGie44) [inline aws/resource_aws_elasticache_replication_group.go:296]: i may have missed this one! prefer just `Sequence`

**Pull request merged**